controllers/metal3.io/host_state_machine_test.go

```diff
@@ -146,6 +146,278 @@ func TestProvisioningCapacity(t *testing.T) {
 	}
 }
 
+func TestDetach(t *testing.T) {
+	testCases := []struct {
+		Scenario                  string
+		Host                      *metal3v1alpha1.BareMetalHost
+		Annotations               map[string]string
```

Comment:
nit: to remove some noise from the case definition, it could be simply something like a `hasDetachAnnotation`

Reply:
Done